.deny.toml

```diff
@@ -0,0 +1,6 @@
+[licenses]
```

Comment:
I'm confused. Do you think that these aren't running? By default, advisory and duplicate dependency checks _are_ running, because `cargo deny check` is an alias for `cargo deny check all`. You can observe this is the [output of the most recent CI check](https://github.com/gfx-rs/wgpu/actions/runs/3414324338/jobs/5682132500) for this PR.

> [advisories] on warn, [duplicate dependencies] on deny.

I have questions!

* The `advisories` section has no top-level policy; it requires more granularity. I think [the defaults](https://embarkstudios.github.io/cargo-deny/checks/advisories/cfg.html) make sense for most projects; paraphrasing below:

    ```toml
	vulnerability = "deny"
	unmaintained = "warn"
	unsound = "warn"
	yanked = "warn"
	notice = "warn"
    ```

    Do you think these defaults are good enough?
* For duplicate dependencies (the [`bans` section](https://embarkstudios.github.io/cargo-deny/checks/bans/index.html)), setting to `"deny"` right now would hard-block even this PR. We currently have some duplicate dependencies with several `{build,dev}-dependencies` for `wgpu`, plus another in `run-wasm`, but not in built code. You can see this with [current CI output](https://github.com/gfx-rs/wgpu/actions/runs/3414324338/jobs/5682132500).

    RE: deny vs. warn here: are potential security issues not a bigger potential problem than duplicated deps? Thinking laterally, I guess we could have duplicate crate checks fail in CI, but not actually make them a required check for PRs. That way, they'd be visible at a PR level. That actually sounds nice.

Reply:
Actually, I don't think there's a meaningful way to separate `bans` checks without gymnastics with multiple configurations. Maybe the best way to resolve it is to allow specific subtrees with duplicates we're willing to tolerate?

Reply:
Does cargo deny bans have no option for ignoring dev deps?

Reply:
> I'm concerned about "sudden" failures in CI due to a new advisory being filed; those sorta suck, and I'd definitely be open to making advisories a separate check, as [suggested in upstream's GH Action for cargo-deny](https://github.com/EmbarkStudios/cargo-deny-action#recommended-pipeline-to-avoid-sudden-breakages).

That would be fine.

> For duplicate dependencies (the [bans section](https://embarkstudios.github.io/cargo-deny/checks/bans/index.html)), setting to "deny" right now would hard-block even this PR. We currently have some duplicate dependencies with several {build,dev}-dependencies for wgpu, plus another in run-wasm, but not in built code. You can see this with [current CI output](https://github.com/gfx-rs/wgpu/actions/runs/3414324338/jobs/5682132500).

What we fundamentally care about is regressions here, so we can add duplicates to the "ignore" section. This is what I do for my projects: https://github.com/BVE-Reborn/rend3/blob/trunk/deny.toml

Reply:
@cwfitzgerald: I've pushed up b2d8a2f2 with a `bans` configuration that:
1. Disallows `multiple-versions` and `wildcards`.
2. Uses `skip-tree`s on DevOps- or debugging-specific crates that aren't currently being published (N.B the addition of c32f51e0 also).

I think that this resolves remaining concerns here (disallowing dupes for consumers of `wgpu`)?

Reply:
Was this properly pushed, I don't see it in the diff?

Reply:
Ah, derp, it wasn't. 😳 Edited my previous comment, pushed.